tools/install.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# Copyright 2020 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+"""Install the parts of emscripten needed for end users. This works like
+a traditional `make dist` target but is writtin python to it can be portable
+and run on non-unix platforms (basically windows).
+"""
+
+from __future__ import print_function
+
+import argparse
+import fnmatch
+import os
+import shutil
+import subprocess
+import sys
+
+EXCLUDES = '''
+tests/third_party
+site
+node_modules
+docs
+Makefile
+.git
+'''.split()
+
+EXCLUDE_PATTERNS = '''
+*.pyc
+.*
+'''.split()
+
+EXCLUDE_DIRS = '''
+__pycache__
+'''.split()
+
+
+def add_revision_file(target):
+  # text=True would be better than encoding here, but its only supported in 3.7+
+  git_hash = subprocess.check_output(['git', 'rev-parse', 'HEAD'], encoding='utf-8').strip()
+  with open(os.path.join(target, 'emscripten-revision.txt'), 'w') as f:
+    f.write(git_hash + '\n')
+
+
+def copy_emscripten(target):
+  script_dir = os.path.dirname(os.path.abspath(__file__))
+  emscripten_root = os.path.dirname(script_dir)
+  os.chdir(emscripten_root)
```

Comment:
why the `chdir` and not just `os.waslk(emscripten_root)`

Reply:
Do that the first element of the return value is already relative to the emscripten root.  Otherwise the first element would be an absolute path and we'd need to convert it to and emscripten-relative path.. I seem to end up doing it this way every time I use `os.walk` which mean its either badly designed or I'm using it wrong :)